IPython/core/ultratb.py

```diff
@@ -336,13 +336,22 @@ def _format_exception_only(
                     )
                 )
                 if textline == "":
-                    # sep 2025:
-                    # textline = py3compat.cast_unicode(value.text, "utf-8")
-                    if value.text is None:
+                    # Gracefully coerce SyntaxError.text into a printable str.
+                    value_text = getattr(value, "text", None)
+                    if value_text is None:
                         textline = ""
                     else:
-                        assert isinstance(value.text, str)
-                        textline = value.text
+                        try:
+                            # Handle bytes by decoding to utf-8, fallback to str() for other types
+                            if isinstance(value_text, bytes):
+                                textline = value_text.decode("utf-8", "replace")
+                            else:
+                                textline = str(value_text)
+                        except Exception:
+                            # As a last resort, fall back to ``str`` for exotic objects.
+                            textline = str(value_text)
+                        if textline is None:
```

Comment:
I'm not sure how textline can be None, you are already in a if textline == ""

Yo need to review your logic or explain how it can be None. 

`str(..)` cannot:

```
In [1]: class Foo:
   ...:     def __str__(self):
   ...:         return None
   ...:

In [2]: str(Foo())
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[2], line 1
----> 1 str(Foo())

TypeError: __str__ returned non-string (type NoneType)
```


I don't think .decode() can.